populate-the-supabase/action.js

```diff
@@ -20,71 +35,103 @@ async function run() {
-        const {contributors_oneGraph} = persistedData.data.gitHub.repositoryOwner.repository;
-
-        const contributorNames = await fetchContributorNames(contributors_oneGraph.nodes)
-
-        item.id = id
-
-        await supabase.from('user_stars').insert({
-          user_id: repository.owner.id,
-          star_id: item.id,
-          repo_name: item.full_name,
-          recency_score: parsedData.indexOf(item),
-          description: description,
-          issues: open_issues,
-          stars: stargazers_count,
-          contributors: contributorNames.slice(0,2) // grab first two names only
-        })
-      }
+    if ((parsedCache[repository.owner.login] !== undefined
+        && parsedCache[repository.owner.login].offsetDays < limitDays)) {
+      continue
+    }
 
-     // send parsedData to stars table
-     await supabase.from('stars').upsert(parsedData  )
+    if (promises.length >= limitUsers) {
+      break
+    }
 
-      console.log(`ADDED STARS FROM: ${repository.html_url}`)
+    promises.push(new Promise(async (resolve, reject) => {
+      try {
+        console.log(`Processing ${repository.full_name}`)
+
+        const {data} = await octokit.rest.repos.getContent({
+          owner: repository.owner.login,
+          repo: repository.name,
+          path: "stars.json"
+        }).catch((err) => {
+          console.log(`stars.json: ${err}`)
+          return {data: {content: []}}
+        })
+
+        // convert from base64 to parseable JSOON
+        const content = Buffer.from(data.content, "base64").toString()
+        const parsedData = JSON.parse(content)
+
+        // update data with repo id
+        for (const item of parsedData) {
+          const [owner, repo] = item.full_name.split("/")
+          const currentRepoResponse = await octokit.rest.repos.get({owner, repo})
+          const {
+            id,
+            stargazers_count,
+            description,
+            open_issues,
+          } = currentRepoResponse.data
+
+          const {data, errors} = await api.persistedRepoDataFetch({owner: owner, repo: repo})
+
+          if (errors && errors.length > 0) {
+            continue
+          }
+
+          const {contributors_oneGraph} = data.gitHub.repositoryOwner.repository;
+
+          const contributorNames = await fetchContributorNames(contributors_oneGraph.nodes)
+
+          item.id = id
+
+          await supabase.from('user_stars').insert({
+            user_id: repository.owner.id,
+            star_id: item.id,
+            repo_name: item.full_name,
+            recency_score: parsedData.indexOf(item),
+            description: description,
+            issues: open_issues,
+            stars: stargazers_count,
+            contributors: contributorNames.slice(0,2) // grab first two names only
+          })
+        }
+
+        // send parsedData to stars table
+        // await supabase.from('stars').upsert(parsedData)
```

Comment:
```suggestion
        await supabase.from('stars').upsert(parsedData)
```
This will need to be uncommented I believe.

Reply:
Yup, still working out some errors - stuck on the alternate supabase db suggestion :D